metadata_service/proxy/neo4j_proxy.py

```diff
@@ -236,7 +241,19 @@ def _exec_table_query(self, table_uri: str) -> Tuple:
             src = Source(source_type=table_records['src']['source_type'],
                          source=table_records['src']['source'])
 
-        return wmk_results, table_writer, timestamp_value, owner_record, tags, src, badges
+        prog_descriptions = self._extract_programmatic_descriptions_from_query(table_records)
```

Comment:
Suggestion: How about passing `prog_descriptions` , an output from `table_records.get('prog_descriptions', [])`, as a method parameter? Just to decouple `table_records` structure from `_extract_programmatic_descriptions_from_query` method.

e.g: _extract_programmatic_descriptions_from_query(table_records.get('prog_descriptions', []))

Reply:
good idea